motor/discovery/aws/ecs.go

```diff
@@ -0,0 +1,335 @@
+	ecsservice "github.com/aws/aws-sdk-go-v2/service/ecs"
+	"github.com/aws/aws-sdk-go-v2/service/ecs/types"
+	ecstypes "github.com/aws/aws-sdk-go-v2/service/ecs/types"
+
+	"go.mondoo.com/cnquery/motor/asset"
+	"go.mondoo.com/cnquery/motor/discovery/common"
+	"go.mondoo.com/cnquery/motor/motorid/awsec2"
+	"go.mondoo.com/cnquery/motor/motorid/containerid"
+	"go.mondoo.com/cnquery/motor/platform"
+	"go.mondoo.com/cnquery/motor/providers"
+	aws_provider "go.mondoo.com/cnquery/motor/providers/aws"
+	"go.mondoo.com/cnquery/resources/library/jobpool"
+	"go.mondoo.com/cnquery/stringx"
+)
+
+func NewECSContainersDiscovery(cfg aws.Config) (*ECSContainers, error) {
+	return &ECSContainers{config: cfg}, nil
+}
+
+type ECSContainers struct {
+	config aws.Config
+}
+
+func (ecs *ECSContainers) Name() string {
+	return "AWS ECS Discover"
+}
+
+func (ecs *ECSContainers) List() ([]*asset.Asset, error) {
+	identityResp, err := aws_provider.CheckIam(ecs.config)
+	if err != nil {
+		return nil, err
+	}
+
+	account := *identityResp.Account
+
+	instances := []*asset.Asset{}
+	poolOfJobs := jobpool.CreatePool(ecs.getECSContainers(account), 5)
+	poolOfJobs.Run()
+
+	// check for errors
+	if poolOfJobs.HasErrors() {
+		return nil, poolOfJobs.GetErrors()
+	}
+	// get all the results
+	for i := range poolOfJobs.Jobs {
+		instances = append(instances, poolOfJobs.Jobs[i].Result.([]*asset.Asset)...)
+	}
+
+	return instances, nil
+}
+
+func (ecs *ECSContainers) getRegions() ([]string, error) {
+	regions := []string{}
+
+	ec2svc := ec2.NewFromConfig(ecs.config)
+	ctx := context.Background()
+
+	res, err := ec2svc.DescribeRegions(ctx, &ec2.DescribeRegionsInput{})
+	if err != nil {
+		return regions, err
+	}
+	for _, region := range res.Regions {
+		regions = append(regions, *region.RegionName)
+	}
+	return regions, nil
+}
+
+func (ecs *ECSContainers) getECSContainers(account string) []*jobpool.Job {
+	tasks := make([]*jobpool.Job, 0)
+	var err error
+
+	regions, err := ecs.getRegions()
+	if err != nil {
+		return []*jobpool.Job{{Err: err}} // return the error
+	}
+	log.Debug().Msgf("regions being called for ecs container instances list are: %v", regions)
+	for ri := range regions {
+		region := regions[ri]
+		f := func() (jobpool.JobResult, error) {
+			// get client for region
+			clonedConfig := ecs.config.Copy()
+			clonedConfig.Region = region
+			containerInstancesAssets := []*asset.Asset{}
+			containersAssets := []*asset.Asset{}
+			svc := ecsservice.NewFromConfig(clonedConfig)
+			ctx := context.Background()
+
+			resp, err := svc.ListClusters(ctx, &ecsservice.ListClustersInput{})
+			if err != nil {
+				return nil, errors.Wrap(err, "could not gather ecs cluster information")
+			}
+			for _, cluster := range resp.ClusterArns {
+				clusterName := ""
+				clusterDescription, err := svc.DescribeClusters(ctx, &ecsservice.DescribeClustersInput{Clusters: []string{cluster}})
+				if err != nil {
+					return nil, errors.Wrap(err, "could not gather ecs cluster information")
```

Comment:
Missed this one before:

When AWS returns multiple clusters, but we get an error on the first one, we would skip all the other clusters.
Perhaps, just log and continue?